test/jsonapi/serializer_test.exs

```diff
@@ -335,13 +335,15 @@ defmodule JSONAPI.SerializerTest do
       ]
     }
 
-    conn = %Plug.Conn{
-      assigns: %{
-        jsonapi_query: %Config{
-          include: [best_comments: :user]
+    conn =
+      %Plug.Conn{
+        assigns: %{
+          jsonapi_query: %Config{
+            include: [best_comments: :user]
+          }
         }
       }
-    }
+      |> Plug.Conn.fetch_query_params()
```

Comment:
Why is this needed?  Is there an assertion we could add for a similar example you had in the PR description?

Reply:
>  Why is this needed?

Because otherwise the `query_params` field of the connection is a `Plug.Conn.Unfetched` struct and `JSONAPI.View.url_for_pagination/3` fails with the following error.

```
1) test with a dasherized API handles sparse fields properly (JSONAPITest)
     test/jsonapi_test.exs:328
     ** (Protocol.UndefinedError) protocol Enumerable not implemented for %{:__struct__ => Plug.Conn.Unfetched, :aspect => :query_params, "page" => %{}} of type Plug.Conn.Unfetched (a struct). This protocol is implemented for the following type(s): HashSet, Range, Map, Function, List, Stream, Date.Range, HashDict, GenEvent.Stream, MapSet, File.Stream, IO.Stream
     code: |> MyPostPlug.call([])
     stacktrace:
       (elixir 1.10.2) lib/enum.ex:1: Enumerable.impl_for!/1
       (elixir 1.10.2) lib/enum.ex:141: Enumerable.reduce/3
       (elixir 1.10.2) lib/enum.ex:3383: Enum.flat_map/2
       test/jsonapi_test.exs:14: JSONAPITest.PostView.url_for_pagination/3
       (jsonapi 1.2.3) lib/jsonapi/serializer.ex:178: JSONAPI.Serializer.merge_links/7
       test/jsonapi_test.exs:86: JSONAPITest.MyPostPlug.passthrough/2
       test/jsonapi_test.exs:73: JSONAPITest.MyPostPlug.plug_builder_call/2
       test/jsonapi_test.exs:334: (test)
```

>  Is there an assertion we could add for a similar example you had in the PR description?

I added a test to `serializer_test.exs`. Does it look good?

Reply:
Great test addition!